stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt

```diff
@@ -274,6 +277,54 @@ internal class ChannelLogic(
         .maxOrNull()
         ?: NEVER.time
 
+    /**
+     * Returns instance of [WatchChannelRequest] to obtain older messages of a channel.
+     *
+     * @param limit Message limit in this request.
+     * @param baseMessageId Message id of the last available message. Request will fetch messages older than this.
+     */
+    internal fun olderWatchChannelRequest(limit: Int, baseMessageId: String?): WatchChannelRequest =
+        watchChannelRequest(Pagination.LESS_THAN, limit, baseMessageId)
+
+    /**
+     * Returns instance of [WatchChannelRequest] to obtain newer messages of a channel.
+     *
+     * @param limit Message limit in this request.
+     * @param baseMessageId Message id of the last available message. Request will fetch messages newer than this.
+     */
+    internal fun newerWatchChannelRequest(limit: Int, baseMessageId: String?): WatchChannelRequest =
+        watchChannelRequest(Pagination.GREATER_THAN, limit, baseMessageId)
+
+    /**
+     * Creates instance of [WatchChannelRequest] according [Pagination].
+     *
+     * @param pagination Pagination parameter which defines should we request older/newer messages.
+     * @param limit Message limit in this request.
+     * @param baseMessageId Message id of the last available. Can be null then it calculates the last available message.
+     */
+    private fun watchChannelRequest(pagination: Pagination, limit: Int, baseMessageId: String?): WatchChannelRequest {
+        val messageId = baseMessageId ?: getLoadMoreBaseMessageId(pagination)
+        return QueryChannelPaginationRequest(limit).apply {
+            messageId?.let {
+                messageFilterDirection = pagination
+                messageFilterValue = it
+            }
+        }.toWatchChannelRequest(chatDomainImpl.userPresence)
+    }
+
+    /** Calculates base messageId for [WatchChannelRequest] depending on [Pagination] when requesting more messages. */
```

Comment:
This doc piece seems to be out of place - formatting and structure wise! Could we clean it up? :]